packages/editor/src/components/block-list/block.js

```diff
@@ -769,6 +773,7 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
 } );
 
 export default compose(
+	pure,
```

Comment:
This is useful to avoid rerendering all the blocks when we add/remove a block in the block list (parent component).

I was also tempted to replace it with `React.memo` as it may be more optimized than our HoC. It would be good to compare performance of these two.